### PR TITLE
Add oracle support to db exploration using sqlalchemy

### PR DESCRIPTION
**Pull request by achauve**
First step for #7 and #26

### CODE REVIEW

**Comment by simonvadee** [comment]
Using sqlalchemy is definitely the way to go! I'll keep an eye on this PR, don't hesitate to assign us to review or ping us on slack (the documentation is clearly underdeveloped...)

File: pagai/services/database_explorer.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def get_sql_url(db_handler: str, sql_config: Dict) -> str:
+    return (
+        f'{db_handler}://{sql_config["login"]}:{sql_config["password"]}@{sql_config["host"]}:{sql_config["port"]}'
+        f'/{sql_config["database"]}'
+    )
+
+
+class DatabaseExplorer:
```
Comment (simonvadee) [inline pagai/services/database_explorer.py]: This is a million times better, thanks!

File: test/test_explore_multi_db.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+from pagai.services.database_explorer import DatabaseExplorer, POSTGRES
+
+
+class TestExploration:
+    def test_explore(self, db_config):
+        explorer = DatabaseExplorer(db_config['handler'], db_config)
+        exploration = explorer.explore(table='patients', limit=2)
+
+        from pprint import pprint
```
Comment (simonvadee) [inline test/test_explore_multi_db.py]: is there a reason why this import isn't at the global scope?

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,4 @@
-FROM python:3.7-slim
-
-RUN apt-get clean \
-    && apt-get -y update
-
-RUN apt-get -y install python3-dev \
-    && apt-get -y install build-essential 
+FROM arkhn/python-base:latest
```
[the remark this answers is not in the document]
Comment (achauve) [reply Dockerfile]: yes but do you want to do it now alongside this PR or later?

File: Dockerfile.base [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+# Dependencies
+RUN apt-get update \
+    && apt-get -y  --no-install-recommends install build-essential python3-dev postgresql python-psycopg2 libpq-dev libaio1 wget unzip \
+    && apt-get autoremove -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install the oracle client
+RUN wget https://download.oracle.com/otn_software/linux/instantclient/195000/instantclient-basic-linux.x64-19.5.0.0.0dbru.zip \
```
[the remark this answers is not in the document]
Comment (achauve) [reply Dockerfile.base]: yes of course

File: docker-compose.test.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    networks:
+      - arkhn
+
+  pagai:
+    build:
+      context: .
+      dockerfile: Dockerfile.test
+    # keep container alive for test commands with `docker-compose exec`
+    command: tail -F anything
+    working_dir: /app
```
[the remark this answers is not in the document]
Comment (achauve) [reply docker-compose.test.yml]: not at all! :) I tried multiple solutions before being satisfied, so this is a residual :D I delete it

File: docker-compose.test.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    volumes:
+      - .:/app
+      - ./models:/srv/pagai/models
+    networks:
+      - arkhn
+
+volumes:
+  postgres-data:
+
+networks:
```
[the remark this answers is not in the document]
Comment (achauve) [reply docker-compose.test.yml]: ok to be honests I copied it from the original docker-compose. What would you do? rename it to a "test" network to not interfere with other postgres instances on local (dev) machine?

File: pagai/services/database_explorer.py (hunk 1 above)
Comment (achauve) [reply pagai/services/database_explorer.py]: Thx :)

File: test/test_explore_multi_db.py (hunk 2 above)
Comment (achauve) [reply test/test_explore_multi_db.py]: haha no it's debug code I forgot to delete.

File: docker-compose.test.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (achauve) [reply docker-compose.test.yml]: actually I'm not sure it is useful to create a custom network at all, don't you think?

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (simonvadee) [reply Dockerfile]: nope ;)

File: docker-compose.test.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (simonvadee) [reply docker-compose.test.yml]: no I don't think it's useful in that case, you can safely remove it

File: docker-compose.test.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (simonvadee) [reply docker-compose.test.yml]: :+1:

**Comment by achauve** [comment]
yes of course, done! (at first I wanted to do multiple PRs but as it was small at the end everything is on the same).

**Pull request merged**